src/views/components/UserProjectRolesTable.vue

```diff
@@ -1,154 +1,318 @@
+            v-for="option in paginationOptions"
+            :key="option"
+            @click="perPage = option"
+            :active="perPage === option"
+          >
+            {{ option }}
+          </b-dropdown-item>
+
+          <b-dropdown-item @click="perPage = 0" :active="perPage === 0">{{
+            $t('admin.pagination_all')
+          }}</b-dropdown-item>
+        </b-dropdown>
+        <span v-if="perPage !== 0" class="text-center">{{
+          currentPageText
+        }}</span>
+      </div>
+      <b-pagination
+        :style="{ visibility: perPage === 0 ? 'hidden' : 'visible' }"
+        v-model="currentPage"
+        :per-page="perPage"
+        :total-rows="rows"
+        first-number
+        class="m-0 h-100"
+      ></b-pagination>
+    </div>
 
     <select-project-modal
-      :username="parentContext.row.username"
+      :username="username"
       @selection="createPrototypeMapping"
     />
   </div>
 </template>
 
 <script>
+/**
+ * Props for UserProjectRolesTable
+ *
+ * @prop {String} username - Required. The username of the user whose project roles are being managed.
+ * @prop {Array<Object>} projectRoles - Required. List of project-role mappings for the user.
+ *   Each item should have the following structure:
+ *   {
+ *     user: {
+ *       username: String, // The username of the user
+ *       // ...other user fields
+ *     },
+ *     project: {
+ *       uuid: String,     // Unique project identifier
+ *       name: String,     // Project name
+ *       // ...other project fields (e.g., active, classifier, etc.)
+ *     },
+ *     role: {
+ *       uuid: String,     // Unique role identifier
+ *       name: String,     // Role name
+ *       permissions: Array, // (optional) List of permissions
+ *       // ...other role fields
+ *     }
+ *   }
+ * @prop {Array<Object>} availableRoles - Required. List of available roles to assign.
+ *   Each item should have the following structure:
+ *   {
+ *     uuid: String,         // Unique role identifier
+ *     name: String,         // Role name
+ *     permissions: Array<{  // (optional) List of permissions for the role
+ *       name: String,       // Permission name
+ *       description: String // Permission description
+ *     }>
+ *     // ...other role fields
+ *   }
+ * @prop {Object} tableOptions - Optional. Table configuration options (pagination, filtering, etc).
+ *   - paginationOptions: Number or array of numbers for per-page dropdown (optional)
+ *      ignored if showPagination is false
+ *   - showPagination: Boolean, show/hide pagination (optional)
+ *   - tableFilter: String, initial filter value (optional)
+ *   - inlineSearch: Boolean, enable inline search (optional)
+ */
+
 import Multiselect from 'vue-multiselect';
 import SelectProjectModal from '../portfolio/projects/SelectProjectModal.vue';
 import i18n from '../../i18n';
+import _ from 'lodash';
+
+const defaultTableOptions = {
+  paginationOptions: [5, 10, 25],
```

Comment:
```suggestion
  paginationOptions: [10, 25, 50],
```
To align with Permissions pagination options